language/libra-vm/src/counters.rs

```diff
@@ -1,174 +1,102 @@
 // Copyright (c) The Libra Core Contributors
 // SPDX-License-Identifier: Apache-2.0
 
-use libra_metrics::{IntCounter, IntGauge, OpMetrics};
-use libra_types::{
-    transaction::TransactionStatus,
-    vm_error::{StatusCode, StatusType, VMStatus},
+use libra_metrics::{
+    register_histogram, register_int_counter_vec, register_int_gauge, Histogram, IntCounterVec,
+    IntGauge,
 };
 use once_cell::sync::Lazy;
-use std::{convert::TryFrom, time::Instant};
 
-// constants used to create counters
-const TXN_EXECUTION_KEEP: &str = "txn.execution.keep";
-const TXN_EXECUTION_DISCARD: &str = "txn.execution.discard";
-const TXN_VERIFICATION_SUCCESS: &str = "txn.verification.success";
-const TXN_VERIFICATION_FAIL: &str = "txn.verification.fail";
-const TXN_BLOCK_COUNT: &str = "txn.block.count";
-pub const TXN_TOTAL_TIME_TAKEN: &str = "txn_gas_total_time_taken";
-pub const TXN_VERIFICATION_TIME_TAKEN: &str = "txn_gas_verification_time_taken";
-pub const TXN_VALIDATION_TIME_TAKEN: &str = "txn_gas_validation_time_taken";
-pub const TXN_EXECUTION_TIME_TAKEN: &str = "txn_gas_execution_time_taken";
-pub const TXN_PROLOGUE_TIME_TAKEN: &str = "txn_gas_prologue_time_taken";
-pub const TXN_EPILOGUE_TIME_TAKEN: &str = "txn_gas_epilogue_time_taken";
-pub const TXN_EXECUTION_GAS_USAGE: &str = "txn_gas_execution_gas_usage";
-pub const TXN_TOTAL_GAS_USAGE: &str = "txn_gas_total_gas_usage";
+/// Count the number of transactions verified, with a "status" label to
+/// distinguish success or failure results.
+pub static TRANSACTIONS_VERIFIED: Lazy<IntCounterVec> = Lazy::new(|| {
+    register_int_counter_vec!(
+        "libra_vm_transactions_verified",
+        "Number of transactions verified",
+        &["status"]
+    )
+    .unwrap()
+});
 
-// the main metric (move_vm)
-pub static VM_COUNTERS: Lazy<OpMetrics> = Lazy::new(|| OpMetrics::new_and_registered("move_vm"));
+/// Count the number of transactions executed, with a "status" label to
+/// distinguish completed vs. discarded transactions.
+pub static TRANSACTIONS_EXECUTED: Lazy<IntCounterVec> = Lazy::new(|| {
+    register_int_counter_vec!(
+        "libra_vm_transactions_executed",
+        "Number of transactions executed",
+        &["status"]
+    )
+    .unwrap()
+});
 
-static VERIFIED_TRANSACTION: Lazy<IntCounter> =
-    Lazy::new(|| VM_COUNTERS.counter(TXN_VERIFICATION_SUCCESS));
-static BLOCK_TRANSACTION_COUNT: Lazy<IntGauge> = Lazy::new(|| VM_COUNTERS.gauge(TXN_BLOCK_COUNT));
+pub static BLOCK_TRANSACTION_COUNT: Lazy<IntGauge> = Lazy::new(|| {
```

Comment:
fwiw this did not give much info when we first added it. I am pointing out just for info and not for any change.
It was the case most block had 0 size and so it was difficult to get info out of this

Reply:
OK, let's get these up on a dashboard so we can see what is useful. I'm surprised that you saw most blocks having 0 size.